### PR TITLE
Don't leak a layer's signature-default call-context arg to sibling layers

### PR DESCRIPTION
**Pull request by pctablet505**
`Layer._resolve_and_populate_arg` resolves a call-context arg (e.g. `training`) by priority: explicit > inherited-from-outer > the layer's own `call()` signature default. `CallContext` is shared across the whole call tree, and the resolved value was always stashed and injected — including the signature default.

So a layer with a non-`None` default (e.g. `Resizing`/`CenterCrop`, whose `call` defaults `training=True`) leaked that default to every later sibling/downstream layer. A `Resizing` upstream of a `BatchNormalization` put the BN in training mode at inference — wrong outputs, plus an in-place moving-stat update that breaks `torch.export`.

Fix: a signature default stays local — only explicit and inherited values propagate. The bound `call()` still applies its own default, so the layer is unchanged, and training during `fit` is preserved (`operation_fn` injects `training` explicitly).

Root cause of keras-team/keras-hub#2495 for DeepLabV3/SegFormer (companion: keras-team/keras-hub#2776). Adds a regression test (tf/jax/torch/numpy).

## Contributor Agreement

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

File: keras/src/layers/layer.py [hunk 1]
```diff
@@ -1068,22 +1068,28 @@ def _resolve_and_populate_arg(
...
+        # (e.g. a preprocessing layer's `training=True`) leak to sibling and
+        # downstream layers. The bound `call()` still applies the default here,
+        # so only explicit (1) and inherited (2) values are propagated.
+        if propagate:
+            call_context.set_value(arg_name, value)
+            if (
+                self._call_has_context_arg.get(arg_name, False)
+                and value is not None
+            ):
+                kwargs[arg_name] = value
```
Comment (hertschuh) [inline keras/src/layers/layer.py]: Thanks for catching and fixing this.

While your fix produces the correct result, I'd like to refactor the code a little bit to make it less confusing.

The `propagate` flag conflates 2 things:
1. the fact that `value` should be saved in `call_context`
2. the fact that `value` should be passed to the `call` method via `kwargs`

- There is actually only one case for 1. It's the "user explicitly passed it" case. So move this `call_context.set_value(arg_name, value)` line 1071 instead of `propagate = True`. Because in the "inherited from outer layer" case, you're simply writing back the value you just extracted, so it's a no-op, you don't need it at all.
- Lines 1077-1078 are now dead code, you retrieve the `value`, but because of the `if propagate` you don't use that value. But to simplify the code coming after, simply do `else: return` and keep the comment
- Because of the `return` you don't need the `propagate` flag and the `if`. You can simply have:
```python
       if (
            self._call_has_context_arg.get(arg_name, False)
            and value is not None
        ):
            kwargs[arg_name] = value
```

**Review by hertschuh — CHANGES_REQUESTED** [review]

**Review by hertschuh — APPROVED** [review]
Thanks for the fix!
